prime-router/src/main/kotlin/cli/helpers/HL7DiffHelper.kt

```diff
@@ -73,20 +71,53 @@ class HL7DiffHelper {
                         )
                         continue
                     }
-                    if (outputFields.size > inputFields.size) {
-                        outputFields.foldIndexed(differences) { index, differenceAccumulator, output ->
+                    if (!effectivelyBlank(outputFields, inputFields)) {
```

Comment:
Did you decide not to do this, or was it just missing from your commit? If you decided against it, that's fine, but a comment about your decision here would help me know that it wasn't a miss.

Reply:
Missed this one, will apply

Reply:
Just realized that if we removed the 'continue' in the catch handler, then "outputFields" might not have value, so invoking effectivelyBlank(inputFields, outputFields) will throw up?

Reply:
I think the placement of my comment might have led to a misunderstanding. To clarify, the try/catch doesn't need to change, but I'm saything that afterwards, instead of
```
                    if (!effectivelyBlank(outputFields, inputFields)) {
                        if (outputFields.size > inputFields.size) {
                            ...
                        }
                        inputfields.foldIndexed(differences) {
                            ...
                        }
                    }
```
we can do something like this to reduce the level of nested conditions:
```
                    if (effectivelyBlank(outputFields, inputFields)) {
                        continue
                    }
                    if (outputFields.size > inputFields.size) {
                        ...
                    }
                    inputfields.foldIndexed(differences) {
                        ...
                    }
```